ttnn/cpp/ttnn/operations/moreh/moreh_nll_loss_backward/device/moreh_nll_loss_backward_device_operation.cpp

```diff
@@ -85,12 +85,15 @@ void MorehNllLossBackwardDeviceOperation::validate_on_program_cache_hit(
     validate_inputs(attributes, tensor_args);
 }
 
-MorehNllLossBackwardDeviceOperation::shape_return_value_t MorehNllLossBackwardDeviceOperation::compute_output_shapes(
+MorehNllLossBackwardDeviceOperation::spec_return_value_t MorehNllLossBackwardDeviceOperation::compute_output_specs(
     const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
-    // To calculate the output shape, we need the channel_size. However, the required tensors, target and output_grad,
-    // do not contain the channel_size information.
-    TT_FATAL(false, "moreh_nll_loss_backward not support create output tensors.");
-    return tensor_args.target_tensor.get_shape();
+    if (tensor_args.input_grad_tensor.has_value()) {
+        return {tensor_args.input_grad_tensor->get_tensor_spec()};
+    }
+    return TensorSpec(
+        tensor_args.target_tensor.get_logical_shape(),
+        TensorLayout(
+            tensor_args.target_tensor.get_dtype(), PageConfig(Layout::TILE), operation_attributes.memory_config));
```

Comment:
To my memory, the `input` and `input_grad` have a shape of `[N, C, d1, d2, ...]`, while `target` has a shape of `[N, d1, d2, ...]`, so returning the shape of the `target` seems incorrect.

If you want to return `input_grad`, you need to take `C` as a separate input.

Reply:
Thank you for the clarification, I restored the original behavior.